.github/workflows/continuous-integration.yml

```diff
@@ -485,9 +551,18 @@ jobs:
   testing-reports:
     name: Testing Reports
     runs-on: ubuntu-latest
-    needs: cypress-tests
-    if: ${{ always() && (needs.cypress-tests.result == 'success' || needs.cypress-tests.result == 'failure') }}
+    needs: [cypress-tests-prep, cypress-tests]
+    if: ${{ always() && (needs.cypress-tests-prep.outputs.num_containers > 0) && (needs.cypress-tests.result == 'success' || needs.cypress-tests.result == 'failure') }}
```

Comment:
Is `always()` needed here?

Reply:
I'm not sure, I didn't add it. I can try removing it.

Reply:
I guess the better question, cypress hasn't been flaky (hooray!), but if it were to fail, do you want this to still run or not? It seems like it has a soft "Always()"

Reply:
Yes, we always want this job to run, even if `cypress-tests` fails, because the test results data is saved in BigQuery.